### PR TITLE
Flip enableLegacyExtensions flag defaults

### PR DESCRIPTION
**Pull request by philipp-spiess**
Part of #40460

This PR flips `enableLegacyExtensions` to `true` by default (if not defined otherwise). After this is merged, extensions are disabled by default.

## Test plan

- Tests pass
- The proper page configs load
Screenshot 2022-09-05 at 12 03 18

### CODE REVIEW

File: client/browser/src/shared/platform/inlineExtensionsService.ts [hunk 1]
```diff
@@ -37,7 +37,10 @@ export const shouldUseInlineExtensions = (requestGraphQL: PlatformContext['reque
             }
 
             try {
-                return result.data.site.enableLegacyExtensions
+                const enableLegacyExtensions = result.data.site.enableLegacyExtensions
+                return typeof enableLegacyExtensions === 'undefined'
+                    ? DEFAULT_ENABLE_LEGACY_EXTENSIONS
+                    : enableLegacyExtensions
```
Comment (philipp-spiess) [inline client/browser/src/shared/platform/inlineExtensionsService.ts:43]: I don't know if `result.data.site.enableLegacyExtensions` can be `undefined`. I guess not? But I figured to be better safe than sorry here 😐

File: client/web/src/integration/search.test.ts [hunk 2]
```diff
@@ -238,8 +238,8 @@ describe('Search', () => {
                     await driver.page.waitForSelector('[data-testid="results-info-bar"]')
                     expect(await editor.getValue()).toStrictEqual('foo')
                     // Field value is cleared when navigating to a non search-related page
-                    await driver.page.waitForSelector('a[href="/extensions"]')
-                    await driver.page.click('a[href="/extensions"]')
+                    await driver.page.waitForSelector('a[href="/notebooks"]')
```
Comment (vdavid) [inline client/web/src/integration/search.test.ts:241]: Nice

File: cmd/frontend/registry/api/extensions_test.go [hunk 3]
```diff
@@ -101,6 +103,8 @@ type mockRegistryExtension struct {
 }
 
 func TestGetExtensionByExtensionID(t *testing.T) {
+	enableLegacyExtensions()
```
Comment (vdavid) [inline cmd/frontend/registry/api/extensions_test.go:106]: Not sure how go tests work, so just to be sure: don't we need some cleanup/reset for this setting after the test?

File: cmd/frontend/registry/api/gating.go [hunk 4]
```diff
@@ -18,10 +18,9 @@ func ExtensionRegistryReadEnabled() error {
 
 func ExtensionRegistryWriteEnabled() error {
 	cfg := conf.Get()
-	if cfg.ExperimentalFeatures != nil && cfg.ExperimentalFeatures.EnableLegacyExtensions != nil && *cfg.ExperimentalFeatures.EnableLegacyExtensions == false {
+	if cfg.ExperimentalFeatures == nil || cfg.ExperimentalFeatures.EnableLegacyExtensions == nil || *cfg.ExperimentalFeatures.EnableLegacyExtensions == false {
```
Comment (vdavid) [inline cmd/frontend/registry/api/gating.go:21]: nice

**Review by vdavid — APPROVED** [review]
Looks good to me.
I didn't check if these were _all_ the places where we needed changes, but I assume you searched for "experimentalFeatures" and checked all places, so it should be good :)

**Comment by philipp-spiess** [comment]
I think it should be safe to merge this now (and change the default)

- For dev environments we want people to be testing with the extensions system disabled anyways
- For S2 we've already manually overwritten the flag to `false` (in which case this PR has no effect)
- For dotcom we've already manually overwritten the flag to `true` (in which case this PR has no effect)


cc @vdavid @taras-yemets @olafurpg

**Pull request merged**